### PR TITLE
Improve comment for BASELINE status in TestStatus

### LINKED ISSUE #810: Test reporting feature request: give baseline tag in a comment in TestStatus
I think this feature was present in cime4, but was dropped in cime5: It's helpful if the baseline tag is given in a comment in TestStatus, so rather than just:

```
  PASS SMS_Ld5.f19_g16.IRCP45CLM45BGC.yellowstone_intel.clm-decStart BASELINE
```

we'd have something like:

```
  PASS SMS_Ld5.f19_g16.IRCP45CLM45BGC.yellowstone_intel.clm-decStart BASELINE clm4_5_12_r203
```

and same for FAILs

The same should be done for GENERATE.

### PR DESCRIPTION
**Pull request by jgfouca**
Adds the baseline name

Test suite: code-checker + by-hand
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #810 

User interface changes?: TestStatus changes

Code review: @jedwards4b @billsacks

### CODE REVIEW

File: scripts/lib/CIME/SystemTests/system_tests_common.py [hunk 1]
```diff
@@ -353,9 +353,10 @@ def _compare_baseline(self):
             success, comments = compare_baseline(self._case)
             append_testlog(comments)
             status = TEST_PASS_STATUS if success else TEST_FAIL_STATUS
-            ts_comments = comments if "\n" not in comments else None
+            baseline_name = self._case.get_value("BASECMP_CASE")
+            ts_comments = comments if "\n" not in comments else os.path.dirname(baseline_name)
```
Comment (billsacks) [inline scripts/lib/CIME/SystemTests/system_tests_common.py]: If there are comments, would it work to print the baseline_name in addition to the other comments - e.g., as `"%s %s"%(os.path.dirname(baseline_name), comments)`? That way we'd be guaranteed to see the baseline name, right?

Comment (jgfouca) [inline scripts/lib/CIME/SystemTests/system_tests_common.py]: Good point.

**Comment by jgfouca** [comment]
@billsacks ok, what do you think now?

**Comment by billsacks** [comment]
Looks good now -thanks.

**Pull request merged**